sys/net/gnrc/application_layer/dhcpv6/client_6lbr.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2018-20 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine S. Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <assert.h>
+
+#include "event.h"
+#include "log.h"
+#include "net/dhcpv6/client.h"
+#include "net/ipv6/addr.h"
+#include "net/gnrc.h"
+#include "net/gnrc/ipv6/nib/ft.h"
+#include "net/gnrc/netif/internal.h"
+
+#include "net/gnrc/dhcpv6/client/6lbr.h"
+
+static char _stack[DHCPV6_CLIENT_STACK_SIZE];
+
+static gnrc_netif_t *_find_upstream_netif(void)
+{
+    gnrc_netif_t *netif = NULL;
+
+    if (CONFIG_GNRC_DHCPV6_CLIENT_6LBR_UPSTREAM) {
+        return gnrc_netif_get_by_pid(CONFIG_GNRC_DHCPV6_CLIENT_6LBR_UPSTREAM);
+    }
+    while ((netif = gnrc_netif_iter(netif))) {
+        if (!gnrc_netif_is_6lo(netif)) {
+            LOG_WARNING("DHCPv6: Selecting interface %d as upstream\n",
+                        netif->pid);
+            return netif;
+        }
+    }
+    return NULL;
+}
+
+static void _configure_upstream_netif(gnrc_netif_t *upstream_netif)
+{
+    ipv6_addr_t addr = {
+            .u8 = { 0xfe, 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 }
+        };
+
+    /* set default route to host machine (as set-up in setup_network.sh) */
+    gnrc_ipv6_nib_ft_add(NULL, 0, &addr, upstream_netif->pid, 0);
+    /* set additional link-local address to allow for easy static route
+     * configuration */
+    addr.u8[15] = 2;
+    gnrc_netif_ipv6_addr_add(upstream_netif, &addr, 64, 0);
+}
```

Comment:
Would be good to have a comment on why an additional address is configured here.

Reply:
Like the one in ll 55-56?

Reply:
I don't know what you mean by 'easy static route configuration'. What makes it easy and what is the alternative?
In the other github comment you write this is not needed of the upstream router is configured properly.

Reply:
I have to admit, I just copied this from the UHCP configuration where the upstream router configuration *is* possible. Easy static route configuration means here: have a well-known next-hop link-local address vs. an auto-configured that you first have to find out.

Reply:
See 48af8784d7

Reply:
Thank you!
So this is just to accommodate the upstream router in case of manual configuration.
I suppose on esp with WiFi uplink this is not needed then, I'll try it out!

Reply:
Depends... with my WRT the ESP had problems to configure a default route. It might be however because I have a very ugly hybrid IPv6 configuration at home (since I don't get a prefix by my ISP yet), so it is not "configured properly" ;-).